ocaml/typing/mtype.ml

```diff
@@ -174,77 +299,74 @@ let rec sig_make_manifest sg =
     in
     Sig_modtype(Ident.rename id, newdecl, vis) :: sig_make_manifest rem
 
-let rec make_aliases_absent pres mty =
+let rec make_aliases_absent ?(aliased=false) pres mty =
   match mty with
   | Mty_alias _ -> Mp_absent, mty
   | Mty_signature sg ->
-      pres, Mty_signature(make_aliases_absent_sig sg)
+      let make_item = function
+        | Sig_module(id, pres, md, rs, priv) ->
+          let pres, md = if aliased
+            then Mp_absent, md
+            else
+              let pres, md_type =
+                make_aliases_absent ~aliased:false pres md.md_type
+              in
+              pres, { md with md_type }
+          in
+          Sig_module(id, pres, md, rs, priv)
+        | item -> item
```

Comment:
Well, this was one spot where I couldn't resist :-) I'll see if changing this back reduces the LOC in the commit. We have no plans to upstream this soon, FWIW.

Reply:
Well, I also like the new version better, of course - I think it's fine to keep it.

As for fragile matches: A general problem throughout the compiler is that it was mostly written before people had any sense of what good OCaml style was, because the language hardly existed.  This creates some tension: should we match the old "style" for consistency or should we write code that is clear / efficient / in good taste.  We resist the urge to fix up old code locally, because it creates merge conflicts, but we try to write new code with good style.  For example, `flambda2` is compiled entirely with the flag that makes fragile matches an error.

Editing and moving around old code is a borderline case - if in some of these cases you think it's clearer to leave the fragile match because you're just moving around old code and it'll be easiest to resolve merge conflicts if we leave it as similar as possible, that's fine.  But where you're substantially changing existing things or writing new functions, I think we prefer to write in good, modern style even if it's a little inconsistent with the surrounding bits.